src/utils.js

```diff
@@ -249,16 +249,32 @@ export function orderProperties(properties, order) {
   if (!Array.isArray(order)) {
     return properties;
   }
-  if (order.length !== properties.length) {
-    throw new Error(
-      "uiSchema order list length should match object properties length");
+
+  const arrayToHash = arr => arr.reduce((prev, curr) => {
+    prev[curr] = true;
+    return prev;
```

Comment:
I must admit, I have no idea... I suspect optimizations are made at the js engine level (the spread operator uses `Object.assign` under the hood), but you're never sure until you produce some benchmarks... *shudder*